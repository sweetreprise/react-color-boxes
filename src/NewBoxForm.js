import React, {useState} from 'react';
import './NewBoxForm.css'

const NewBoxForm = ({ addBox }) => {
    const INITIAL_STATE = {
        width: '',
        height: '',
        bgColor: '#000000'
    }
    const [formData, setFormData] = useState(INITIAL_STATE);
    const handleChange = (e) => {
        const { name, value } = e.target;
        setFormData(formData => ({
            ...formData,
            [name]: value
        }))
    }

    const handleSubmit = (e) => {
        e.preventDefault();
        addBox(parseInt(formData.width), parseInt(formData.height), formData.bgColor);
        setFormData(INITIAL_STATE);
    }

    return (
        <form onSubmit={handleSubmit}>
            <label htmlFor="width">Width:</label>
            <input 
                id="width"
                type="number" 
                min="100"
                max="500"
                name="width" 
                placeholder="Width"
                value={formData.width}
                onChange={handleChange}/>

            <label htmlFor="height">Height:</label>
            <input 
                id="height"
                type="number" 
                min="100"
                max="500"
                name="height" 
                placeholder="Height"
                value={formData.height}
                onChange={handleChange}/>

            <label htmlFor="bgColor">Background-color:</label>
            <input 
                id="bgColor"
                type="color" 
                name="bgColor" 
                placeholder="background-color"
                value={formData.bgColor}
                onChange={handleChange}/>
            
            <button>Enter</button>
        </form>
    )
}

export default NewBoxForm