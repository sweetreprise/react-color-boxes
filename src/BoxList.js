import React, {useState} from 'react'
import Box from './Box'
import NewBoxForm from './NewBoxForm'
import { v4 as uuid } from 'uuid';

const BoxList = () => {
    const INITIAL_STATE = [
        { id: uuid(), width: 100, height: 100, bgColor: '#000000'}
    ]
    const [boxes, setBoxes] = useState(INITIAL_STATE);
    // const addBox = (width, height, bgColor) => {
    //     setBoxes(boxes => [...boxes, { id: uuid(), width, height, bgColor }])
    // }

    function addBox(width, height, bgColor) {
        setBoxes(boxes => [...boxes, { id: uuid(), width, height, bgColor }])
    }

    return (
        <div>
            <h3>BOXES!</h3>
            <NewBoxForm addBox={addBox}/>
            <div>
                {boxes.map(({id, width, height, bgColor}) => <Box 
                    key={id}
                    width={width}
                    height={height}
                    bgColor={bgColor}
                />)}
            </div>
        </div>
    )
}

export default BoxList;