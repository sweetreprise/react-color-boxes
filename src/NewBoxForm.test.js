import React from 'react';
import { render, fireEvent, getByLabelText } from '@testing-library/react';
import NewBoxForm from './NewBoxForm';
import BoxList from './BoxList'

it("renders component without crashing", () => {
    render(<NewBoxForm />);
});

it("matches snapshot", () => {
    const { asFragment } = render(<NewBoxForm />);
    expect(asFragment()).toMatchSnapshot();
});

it('should add a new box', () => {
    const { queryByText, getByLabelText, getByText } = render(<NewBoxForm />);
    const width = getByLabelText('Width:');
    const height = getByLabelText('Height:');
    const colour = getByLabelText('Background-color:');
    const btn = queryByText('Enter');

    // make sure box we're adding is not in document
    expect(queryByText('#8B0000')).not.toBeInTheDocument();
    
    fireEvent.change(width, {target: {value: '200'}})
    fireEvent.change(height, {target: {value: '250'}})
    fireEvent.change(colour, {target: {value: '#8B0000'}})
    fireEvent.click(btn);

    // expect(queryByText('#8B0000')).toBeInTheDocument();
})

// TO-DO: figure out error