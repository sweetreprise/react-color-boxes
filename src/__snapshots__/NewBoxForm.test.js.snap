// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <form>
    <label
      for="width"
    >
      Width:
    </label>
    <input
      id="width"
      max="500"
      min="100"
      name="width"
      placeholder="Width"
      type="number"
      value=""
    />
    <label
      for="height"
    >
      Height:
    </label>
    <input
      id="height"
      max="500"
      min="100"
      name="height"
      placeholder="Height"
      type="number"
      value=""
    />
    <label
      for="bgColor"
    >
      Background-color:
    </label>
    <input
      id="bgColor"
      name="bgColor"
      placeholder="background-color"
      type="color"
      value="#000000"
    />
    <button>
      Enter
    </button>
  </form>
</DocumentFragment>
`;
